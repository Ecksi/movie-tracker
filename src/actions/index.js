export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies
});

export const toggleLogin = () => ({
  type: 'TOGGLE_LOGIN'
});
