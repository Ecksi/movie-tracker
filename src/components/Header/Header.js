import React from 'react';
import { NavLink } from 'react-router-dom';

export const Header = () => {
  const displayLoggedIn = 
    <div>
      <p>Welcome userName</p>
      <NavLink to='/favorites'>Favorites</NavLink>;
      <NavLink to='/'>Sign Out</NavLink>
    </div>;

  const displayLoggedOut = 
    <div>
      <form action="">
        <input type="text" placeholder="Enter User Name" />
        <input type="text" placeholder="Enter Password" />
        <button>Login</button>
      </form>
      <NavLink to='/signup'>Sign Up</NavLink>;
    </div>;

  return (
    <header>
      <NavLink to='/'>
        <h1>MOVIE TRACKER</h1>
      </NavLink>
    </header>
  );
};
  // { userLoggedIn ? displayLoggedIn : displayLoggedOut }