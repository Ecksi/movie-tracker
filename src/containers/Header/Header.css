.userControls {
  text-align: right;
  position: relative;
  bottom: 120px; }
  .userControls button {
    background: #77b6e0;
    background-image: linear-gradient(to bottom, #77b6e0, #417494);
    border-radius: 5px;
    text-shadow: 1px 1px 3px #666666;
    font-family: Arial;
    color: #ffffff;
    font-size: 20px;
    padding: 10px 20px 10px 20px;
    text-decoration: none;
    margin-right: 1.5rem; }
  .userControls button:hover {
    background: #59a5d4;
    background-image: linear-gradient(to bottom, #59a5d4, #3a84b5);
    cursor: pointer; }
  .userControls p {
    color: lightblue;
    font-size: 1.5rem;
    position: relative;
    top: 15px;
    right: 100px; }
