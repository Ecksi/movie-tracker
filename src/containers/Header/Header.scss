header {
  height: 9rem;
}

.userControls {
  position: relative;
  bottom: 5rem;
  left: 67rem;
  display: flex;
  text-align: right;

  .navLink {
    text-decoration: none;
  }

  .favoritesButton {
    background-image: url('https://fontmeme.com/permalink/180521/3eca9b896f474ebbadf5a6e319ec2756.png');
    background-size: 90%;
  }

  .favoritesButton:hover {
    background-image:url('https://fontmeme.com/permalink/180521/ccdff501cc41a10b3c92255dced867cb.png');
    background-size: 80%;
  }

  .logOutButton {
    background-image: url('https://fontmeme.com/permalink/180521/bd5a24ecf1b03206300104fe5f4018b9.png');
    background-position-y: 80%;
  }

  .logOutButton:hover {
    background-image: url('https://fontmeme.com/permalink/180521/8caf3896f2126801e272c59b6e1f8dee.png');
  }

  button {
    display: flex;
    margin-right: 1.5rem;
  }

  p {
    position: relative;
    top: 55px;
    right: 330px;
    color: lightblue;
    font-family: 'Permanent Marker';
    font-size: 1.5rem;
    letter-spacing: 1px;
  }
}