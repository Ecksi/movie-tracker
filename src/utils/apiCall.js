import apiKey from '../private/apiKey';
import { doFetch}  from './doFetch';

const getMovieData = async () => {
  const url = 'https://api.themoviedb.org/3/movie/now_playing?api_key=';

  return await doFetch(url+apiKey);
};

export default getMovieData;