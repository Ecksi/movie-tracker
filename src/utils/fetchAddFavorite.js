export const fetchAddFavorite = async (movie, user_id) => {
  const url = 'http://localhost:3000/api/users/favorites/new';
  const options = {
    method: 'POST',
    body: JSON.stringify({
      ...movie,
      user_id
    }),
    headers: { 'Content-Type': 'application/json' }
  };
  return await fetch(url, options);
};
