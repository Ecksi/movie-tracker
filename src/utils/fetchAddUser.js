export const fetchAddUser = async user => {
  const url = 'http://localhost:3000/api/users/new';
  const options = {
    method: 'POST',
    body: JSON.stringify(user),
    headers: { 'Content-Type': 'application/json' }
  };
  await fetch(url, options);
};
