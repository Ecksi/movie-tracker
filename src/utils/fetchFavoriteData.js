import { doFetch } from './doFetch';

export const fetchFavorites = async user_id => {
  const url = `http://localhost:3000/api/users/${user_id}/favorites`;
  const favoritesList = await doFetch(url);
  return favoritesList.data;
};

