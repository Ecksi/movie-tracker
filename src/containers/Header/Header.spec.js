import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { toggleLogin, loadFavorites } from '../../actions/';
import { fetchFavoriteData } from '../../utils/fetchFavoriteData';
jest.mock('../../utils/fetchFavoriteData');

describe('Header', () => {
  let header;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      loggedIn: false,
      name: 'mock name',
      userId: 1,
      toggleLogin: jest.fn(),
      loadFavorites: jest.fn()
    };
    header = shallow(<Header {...mockProps} />);
  });

  it('matches snapshot with user loggedIn true', () => {
    const mockState = {
      user: {
        loggedIn: true,
        name: 'mock name',
        id: 1
      }
    };
    const mappedProps = mapStateToProps(mockState);
    header = shallow(
      <Header
        {...mappedProps}
        toggleLogin={jest.fn()}
        loadFavorites={jest.fn()}
      />
    );

    expect(header).toMatchSnapshot();
  });

  it.skip('calls toggleLogin when logout button is click', () => {
    // how to expect event handler was called

    const mockState = {
      user: {
        loggedIn: true,
        name: 'mock name',
        id: 1
      }
    };
    const mappedProps = mapStateToProps(mockState);
    header = shallow(
      <Header
        {...mappedProps}
        toggleLogin={jest.fn()}
        loadFavorites={jest.fn()}
      />
    );

    // console.log(header.debug());

    const logOuts = header.find('.logOut');

    console.log(logOuts.debug());

    console.log(logOuts.simulate('click'));

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  describe('mapStateToProps', () => {
    it('should map state properties to props', () => {
      const mockState = {
        user: {
          loggedIn: false,
          name: 'mock name',
          id: 1
        }
      };
      const expected = {
        loggedIn: false,
        name: 'mock name',
        userId: 1
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
  describe('mapDispatchToProps', () => {
    it('calls dispatch on toggleLogin', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = toggleLogin();
      mappedProps.toggleLogin();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch with the correct params on loadFavorites', () => {
      const mockDispatch = jest.fn();
      const mockFavoritesArray = [];
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = loadFavorites(mockFavoritesArray);
      mappedProps.loadFavorites(mockFavoritesArray);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});
