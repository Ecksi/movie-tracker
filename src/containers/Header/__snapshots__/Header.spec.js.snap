// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot with user loggedIn true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    loadFavorites={[Function]}
    loggedIn={true}
    name="mock name"
    toggleLogin={[Function]}
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/"
>
          <img
                    alt="movie-tracker logo"
                    className="headerLogo"
                    src="https://fontmeme.com/permalink/180520/17ff7fa8ddbb9659c011cbf5dacff735.png"
          />
</NavLink>,
        <div
          className="userControls"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/"
          >
                    <button
                              className="logOut"
                              onClick={[Function]}
                    >
                              Log Out
                    </button>
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/favorites"
          >
                    <button
                              className="favorites"
                              onClick={[Function]}
                    >
                              Favorites
                    </button>
          </NavLink>
          <p>
                    Welcome back, 
                    mock name
          </p>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <img
            alt="movie-tracker logo"
            className="headerLogo"
            src="https://fontmeme.com/permalink/180520/17ff7fa8ddbb9659c011cbf5dacff735.png"
/>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie-tracker logo",
            "className": "headerLogo",
            "src": "https://fontmeme.com/permalink/180520/17ff7fa8ddbb9659c011cbf5dacff735.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/"
>
              <button
                            className="logOut"
                            onClick={[Function]}
              >
                            Log Out
              </button>
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/favorites"
>
              <button
                            className="favorites"
                            onClick={[Function]}
              >
                            Favorites
              </button>
</NavLink>,
            <p>
              Welcome back, 
              mock name
</p>,
          ],
          "className": "userControls",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="logOut"
                onClick={[Function]}
>
                Log Out
</button>,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log Out",
                "className": "logOut",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log Out",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="favorites"
                onClick={[Function]}
>
                Favorites
</button>,
              "to": "/favorites",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorites",
                "className": "favorites",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorites",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome back, ",
                "mock name",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Welcome back, ",
              "mock name",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
>
            <img
                        alt="movie-tracker logo"
                        className="headerLogo"
                        src="https://fontmeme.com/permalink/180520/17ff7fa8ddbb9659c011cbf5dacff735.png"
            />
</NavLink>,
          <div
            className="userControls"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/"
            >
                        <button
                                    className="logOut"
                                    onClick={[Function]}
                        >
                                    Log Out
                        </button>
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/favorites"
            >
                        <button
                                    className="favorites"
                                    onClick={[Function]}
                        >
                                    Favorites
                        </button>
            </NavLink>
            <p>
                        Welcome back, 
                        mock name
            </p>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <img
              alt="movie-tracker logo"
              className="headerLogo"
              src="https://fontmeme.com/permalink/180520/17ff7fa8ddbb9659c011cbf5dacff735.png"
/>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie-tracker logo",
              "className": "headerLogo",
              "src": "https://fontmeme.com/permalink/180520/17ff7fa8ddbb9659c011cbf5dacff735.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/"
>
                <button
                                className="logOut"
                                onClick={[Function]}
                >
                                Log Out
                </button>
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/favorites"
>
                <button
                                className="favorites"
                                onClick={[Function]}
                >
                                Favorites
                </button>
</NavLink>,
              <p>
                Welcome back, 
                mock name
</p>,
            ],
            "className": "userControls",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="logOut"
                  onClick={[Function]}
>
                  Log Out
</button>,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log Out",
                  "className": "logOut",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Log Out",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <button
                  className="favorites"
                  onClick={[Function]}
>
                  Favorites
</button>,
                "to": "/favorites",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Favorites",
                  "className": "favorites",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Favorites",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome back, ",
                  "mock name",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Welcome back, ",
                "mock name",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
